### PR TITLE
updated reduceIntents.rst and added an example

### PR DESCRIPTION
**Pull request by vasslitvinov**
to reflect changes since 1.12.

Added an example of a simple user-defined reduction class.

### CODE REVIEW

**Comment by vasslitvinov** [comment]
@ben-albrecht would you review?

File: doc/release/technotes/reduceIntents.rst [hunk 1]
```diff
@@ -82,6 +89,41 @@ Set ``x`` in the loop -- counts the number of tasks:
   }
   writeln("The number of tasks is: ", x);
 
+For a user-defined reduction, there is a task-private instance
+of the reduction class for each task created for the forall
+or coforall loop. Here is an example of such a class that
+implements + reduction over numeric data.
```
Comment (ben-albrecht) [inline doc/release/technotes/reduceIntents.rst]: Maybe '+' reduction, or `+` reduction here?

**Comment by ben-albrecht** [comment]
I can't comment on the completeness of content coverage, but the changes look good to me with some minor suggestions.

File: doc/release/technotes/reduceIntents.rst [hunk 2]
```diff
@@ -82,6 +89,41 @@ Set ``x`` in the loop -- counts the number of tasks:
...
+    type eltType;
+
+    // task-private accumulator state
+    var value: eltType;
+
+    // identity w.r.t. the reduction operation
+    proc identity         return 0: eltType;
+
+    // accumulate a single element onto the accumulator
+    proc accumulate(elm)  { value = value + elm; }
```
Comment (ben-albrecht) [inline doc/release/technotes/reduceIntents.rst]: Should we lead by example and use chpldoc style comments in our examples?

Comment (vasslitvinov) [inline doc/release/technotes/reduceIntents.rst]: Good suggestion - will do.

**Pull request merged**